### ISSUE TITLE
Move widget stack state to list format

### ISSUE THREAD

**Issue by ryanml** [brave/brave-browser#9584]
## Description 
Currently, the state of the widget stack on the NTP is binary:

```
export type StackWidget = 'rewards' | 'binance'

interface NTPState {
  currentStackWidget: StackWidget
}
```

This doesn't allow for ease of adding more widgets to it, which we plan to do. This attribute should be moved to a list, which is enumerated over and controls the display of the widgets in the stack. This will result in 0 change to the user experience, but will pave the way for the future widgets we plan to add. The piece of work should also migrate older states when upgrading to this new format.

## Test Plan
1. Clean Profile
2. Perform a general regression test of the widget stack, which includes showing/hiding Binance and Rewards, as well as toggling between the two to move them between the foreground and background.
3. Ensure that turning the widgets on and off via `brave://settings/newTab` works as expected.
4. Upgrade from a previous profile, with varying states of the widget stack (ex: each in foreground, one off, one on, both off) and ensure that the state is preserved upon upgrade.

**Comment by srirambv** [comment]
Blocked by #10067

**Comment by kjozwiak** [comment]
As discussed during the desktop planning and triage meeting, https://github.com/brave/brave-browser/issues/10067 will be addressed in another. If it's fixed in time, we can get it into `1.11.x`.

Lets run through as much as we can with the above cases and label this as verified. Removing the `QA/Blocked` label.

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.10.90 Chromium: 83.0.4103.97 (Official Build) (64-bit)
-- | --
Revision | 326d148b9655369b86498d9ecca39f63dd2bdd2d-refs/branch-heads/4103@{# 657}
OS | Linux

<details>
<Summary>Clean Install</Summary>
<h4>Widget Position</h4>
<li>Verified Rewards widget is the first widget on a clean profile and Binance widget is in background</li>
<h4>Toggle</h4>
<li>Verified disabling Binance via dashboard and enabling it brings Binance widget into foreground</li>
<li>Verified with rewards in background and disabling it via dashboard and enabling it again brings rewards into focus </li>
<li>Verified when rewards widget shows grant notification, disabling and enabling doesn't lose the notification</li>
<li>Verified when a rewards notification is shown on the widget, clicking Binance to bring it to focus and then clicking rewards again doesn't lose the notification</li>
<li>Verified `Hide Binance` hides the widget and disables in dashboard. Also removes disconnects widget when hidden</li>
<li> Encountered  #10067 and #10199 during testing</li>
</details>

<details>
<Summary>Upgrade Install</Summary>
<h4>Both widgets enabled</h4>

```
- Clean install `1.9.80`
- Keep both widgets enabled
- Upgraded to `1.10.90`
- Both widgets remained enabled
```

<h4>Both widgets disabled</h4>

```
- Clean install `1.9.80`
- Disabled both widgets from dashboard
- Upgraded to `1.10.90`
- Both widgets remained disabled
```
<h4>Binance enabled Rewards disabled</h4>

```
- Clean install `1.9.80`
- Disable rewards via dashboard and keep Binance enabled
- Upgraded to `1.10.90`
- Rewards remained disabled and Binance remained enabled
```

<h4>Binance disabled Rewards enabled</h4>

```
- Clean install `1.9.80`
- Disable Binance via dashboard and keep Rewards enabled
- Upgraded to `1.10.90`
- Binance remained disabled and Rewards remained enabled
```
<h4>Binance Widget state</h4>
<li> Verified upgrading from `1.9.80` to `1.10.90` retains the widgets connected state</li>

</details>
